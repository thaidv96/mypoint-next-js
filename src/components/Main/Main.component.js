import React from 'react'
import PropTypes from 'prop-types'
import styles from './Main.styles'

const Main = ({ children }) => (
  <main
    className="Main"
  >
    {children}
    <style jsx>{styles}</style>
  </main>
)

Main.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.func,
  ]),
}

Main.defaultProps = {
  children: null,
}

export default Main
