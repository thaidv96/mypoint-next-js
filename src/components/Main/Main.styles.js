import css from 'styled-jsx/css'

export default css`
  .Main {
    mix-blend-mode: color-burn;
    border-width: 1px 0 1px 0;
    border-color: #444;
    color: #333;
    border-style: solid;
    font-weight: lighter;
    line-height: 1.8;
    letter-spacing: 0.1em;
  }
`
