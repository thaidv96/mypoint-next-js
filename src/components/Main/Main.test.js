/* eslint-env jest */
/* global window */
import React from 'react'
import { shallow } from 'enzyme'

import Main from './Main.component'

describe('Main', () => {
  const wrapper = shallow(
    <Main>
      <div className="child" />
      <div className="child" />
    </Main>
  )

  it('matches its snapshots', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders its children', () => {
    const childrenExist = wrapper
      .find('.child')
      .exists()

    expect(childrenExist).toBe(true)
  })
})
