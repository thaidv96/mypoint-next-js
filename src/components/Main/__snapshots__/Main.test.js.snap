// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main matches its snapshots 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main>
    <div
      className="child"
    />
    <div
      className="child"
    />
  </Main>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <div
            className="child"
          />,
          <div
            className="child"
          />,
        ],
        <JSXStyle
          css=".Main.jsx-2008063794{mix-blend-mode:color-burn;border-width:1px 0 1px 0;border-color:#444;color:#333;border-style:solid;font-weight:lighter;line-height:1.8;-webkit-letter-spacing:0.1em;-moz-letter-spacing:0.1em;-ms-letter-spacing:0.1em;letter-spacing:0.1em;}"
          styleId="2008063794"
        />,
      ],
      "className": "jsx-2008063794 Main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "child",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "child",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": ".Main.jsx-2008063794{mix-blend-mode:color-burn;border-width:1px 0 1px 0;border-color:#444;color:#333;border-style:solid;font-weight:lighter;line-height:1.8;-webkit-letter-spacing:0.1em;-moz-letter-spacing:0.1em;-ms-letter-spacing:0.1em;letter-spacing:0.1em;}",
          "styleId": "2008063794",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <div
              className="child"
            />,
            <div
              className="child"
            />,
          ],
          <JSXStyle
            css=".Main.jsx-2008063794{mix-blend-mode:color-burn;border-width:1px 0 1px 0;border-color:#444;color:#333;border-style:solid;font-weight:lighter;line-height:1.8;-webkit-letter-spacing:0.1em;-moz-letter-spacing:0.1em;-ms-letter-spacing:0.1em;letter-spacing:0.1em;}"
            styleId="2008063794"
          />,
        ],
        "className": "jsx-2008063794 Main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "child",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "child",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": ".Main.jsx-2008063794{mix-blend-mode:color-burn;border-width:1px 0 1px 0;border-color:#444;color:#333;border-style:solid;font-weight:lighter;line-height:1.8;-webkit-letter-spacing:0.1em;-moz-letter-spacing:0.1em;-ms-letter-spacing:0.1em;letter-spacing:0.1em;}",
            "styleId": "2008063794",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
